test/output_test_helper.cc

```diff
@@ -31,21 +31,21 @@ TestCaseList& GetTestCaseList(TestCaseID ID) {
 
 SubMap& GetSubstitutions() {
   // Don't use 'dec_re' from header because it may not yet be initialized.
-  static std::string dec_re = "[0-9]*[.]?[0-9]+([eE][-+][0-9]+)?";
+  static std::string dec_re2 = "[0-9]*[.]?[0-9]+([eE][-+][0-9]+)?";
```

Comment:
@dominichamon is probably right. It's easier to fix than it is to suppress. However maybe a name like `safe_dec_re` is more appropriate given the above comment?

However I must add that I think ICC is *INSANE* emitting a -Wshadow diagnostic between a global symbol and a function-local variable. God forbid you ever write `#include <math.h> void foo() { int abs = bar(); }`.